### DISCUSSION TITLE
How to update components using the rest api

### DISCUSSION THREAD

**Discussion by abanchev** [strapi/strapi#8231] [Legacy Topics]
When updating relations using the REST api i am sending the id of the referenced objects:
for one-to-one 
```
{
  related_object: 1
}
```
many to one
```
{
  related_objects: [1,2,3...]
}
```

however with components this does not work, if I have a repeatable object it wants me to send the whole object instead of just the id:
```
{
  my_components:[{id:1, data:''}, {id:2, data:''}...]
}
```

This difference is very bad as some relations NEED to be just id and not the whole object, while the components need to be objects and not ids
and this is causing my data normalization to not work, as there is no way to deduce what kind of data i should send to the rest api (and be sure that the data persists correctly)

**Comment by derrickmehaffy** [comment]
Hello,

I see you are wanting to ask a question, we are currently planning to decomission our GitHub Discussions in early 2021 and we don't want to lose the great questions asked by our community.

Questions should be directed to [our forum](https://forum.strapi.io). Please see the following contributing guidelines for asking a question [here](https://github.com/strapi/strapi/blob/master/CONTRIBUTING.md#reporting-an-issue).
